updater/spec/dependabot/analyse_files_command_spec.rb

```diff
@@ -0,0 +1,456 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+require "dependabot/analyse_files_command"
+require "dependabot/bundler"
+require "tmpdir"
+
+RSpec.describe Dependabot::AnalyseFilesCommand do
+  subject(:job) { described_class.new }
+
+  let(:service) do
+    instance_double(Dependabot::Service,
+                    capture_exception: nil,
+                    mark_job_as_processed: nil,
+                    record_update_job_error: nil,
+                    record_update_job_unknown_error: nil,
+                    update_dependency_list: nil,
+                    increment_metric: nil,
+                    wait_for_calls_to_finish: nil)
+  end
+  let(:job_definition) do
+    JSON.parse(fixture("file_fetcher_output/output.json"))
+  end
+  let(:job_id) { "123123" }
+
+  before do
+    allow(Dependabot::Service).to receive(:new).and_return(service)
+    allow(Dependabot::Environment).to receive_messages(
+      job_id: job_id,
+      job_token: "mock_token",
+      job_definition: job_definition,
+      repo_contents_path: nil
+    )
+  end
+
+  describe "#perform_job" do
+    subject(:perform_job) { job.perform_job }
+
+    it "sends dependency metadata to the service", pending: "TODO: Decide if we should submit the legacy dep list" do
+      expect(service).to receive(:update_dependency_list)
+        .with(dependency_snapshot: an_instance_of(Dependabot::DependencySnapshot))
+
+      perform_job
+    end
+
+    describe "experiment: enable_dependency_submission_poc" do
+      context "when it is enabled" do
+        before do
+          Dependabot::Experiments.register(:enable_dependency_submission_poc, true)
+        end
+
+        after do
+          Dependabot::Experiments.reset!
+        end
+
+        it "emits a create_dependency_submission call to the Dependabot service" do
+          expect(service).to receive(:create_dependency_submission) do |args|
+            expect(args[:dependency_submission]).to be_a(GithubApi::DependencySubmission)
+
+            expect(args[:dependency_submission].job_id).to eql(job_id)
+            expect(args[:dependency_submission].package_manager).to eql("bundler")
+          end
+
+          perform_job
+        end
+      end
+
+      context "when it is disabled" do
+        subject(:perform_job) { job.perform_job }
+
+        before do
+          Dependabot::Experiments.register(:enable_dependency_submission_poc, false)
+        end
+
+        after do
+          Dependabot::Experiments.reset!
+        end
+
+        it "does not emits a create_dependency_submission call to the Dependabot service" do
+          expect(service).not_to receive(:create_dependency_submission)
+
+          perform_job
+        end
+      end
+    end
+  end
+
+  # TODO(brrygrdn): Share these tests with UpdateFilesCommand?
```

Comment:
fwiw, my rule of thumb about these situations is don't abstract it until you have to do it 3 times.

copy and paste without shame.